test/test_networking.py

```diff
@@ -52,3 +63,253 @@ def test_start_server_app_kwargs():
     client = TestClient(app)
     assert client.get(f"{API_PREFIX}/docs").status_code == 200
     io.close()
+
+
+def test_reverse_proxy():
+    import gradio as gr
+
+    demo = Interface(lambda x: x, "text", "text")
+    app = FastAPI()
+
+    gr.mount_gradio_app(app, demo, path="/gradio")
+
+    target_port = 8000
```

Comment:
Do we need to hardcode ports or can we choose them dynamically based on the first available port? I'm worried about ports not being availablle on a particular environment

Reply:
Basically this: https://github.com/gradio-app/gradio/blob/fa5d4339d6aafda529150d9536b895e2b239c831/gradio/http_server.py#L116

Reply:
fixed